lib/facter/consul_version.rb

```diff
@@ -0,0 +1,15 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  setcode do
+    if Facter.value(:kernel) == 'Linux'
+      version = Facter::Util::Resolution.exec('consul --version')
+      version = version.lines.first.split[1].tr('v','')
+    elsif Facter.value(:kernel) == 'windows'
+      version = Facter::Util::Resolution.exec('"C:\\Program Files\\Consul\\consul.exe" --version')
```

Comment:
Agree, not optimal.  Since this module doesn't yet support Windows, maybe best to remove Windows support in the fact?  Only reason it's there currently is because I/we use Consul on Windows but use a separate module to install/manage it.